bundles/ch.elexis.core.findings.util/src/ch/elexis/core/findings/util/fhir/transformer/AppointmentTerminTransformer.java

```diff
@@ -75,15 +76,14 @@ public Optional<IAppointment> updateLocalObject(Appointment fhirObject, IAppoint
 		String originalReason = localObject.getReason();
 		LocalDateTime originalEndTime = localObject.getEndTime();
 		attributeMapper.fhirToElexis(fhirObject, localObject);
-		if (!originalEndTime.equals(localObject.getEndTime())) {
+		if (!Objects.equals(originalEndTime, localObject.getEndTime())) {
 			AppointmentHistoryServiceHolder.get().logAppointmentDurationChange(localObject, originalEndTime,
 					localObject.getEndTime());
-			}
-		if (!originalReason.equals(fhirObject.getDescription())) {
+		}
+		if (!Objects.equals(originalReason, fhirObject.getDescription())) {
```

Comment:
Objects.equals würde auch true liefern wenn `originalEndTime` und `localObject.getEndTime()` null sind. Dann würde in der nächsten Zeile eine NPE geschmissen.

Hier muss separat geprüft werden was nicht null sein darf. Vmtl. beide Variablen.